python/grass/script/tests/test_script_task.py

```diff
@@ -1,11 +1,56 @@
+import os
+
+import pytest
+
+import grass.script as gs
 from grass.script.task import grassTask
 
 
-def test_mapcalc_simple_e_name():
+@pytest.fixture
+def xy_session_patched_env(tmp_path, monkeypatch):
+    """Active session in an XY location (scope: function), patching env vars directly.
+
+    This allows functions not accepting an env dictionnary argument to work in tests"""
+    location = "xy_test"
+    gs.core._create_location_xy(tmp_path, location)  # pylint: disable=protected-access
+    with gs.setup.init(tmp_path / location, env=os.environ.copy()) as session:
+        for key, value in session.env.items():
+            monkeypatch.setenv(key, value)
+        yield session
+
+
+def test_mapcalc_simple_e_name(xy_session_patched_env):
     gt = grassTask("r.mapcalc.simple")
     assert gt.get_param("e")["name"] == "e"
 
 
-def test_mapcalc_simple_expession_name():
+def test_mapcalc_simple_expression_name(xy_session_patched_env):
     gt = grassTask("r.mapcalc.simple")
     assert gt.get_param("expression")["name"] == "expression"
+
+
+def test_d_vect_from_bin(xy_session_patched_env):
+    """Tests that a module installed in "$GISBASE/bin can be used with grassTask"""
+    task = grassTask("d.vect")
```

Comment:
I am wondering if using a `d.*` command in this setting is a good idea - will it not lead to

`ERROR: Neither MONITOR (managed by d.mon command) nor GRASS_RENDER_IMMEDIATE (used for direct rendering) defined`
?

Reply:
If I understand that part correctly, on what grassTask and that code snippet does, is that it plays more with the parser and what options are possible, rather than actually running it.

If it had an error, I think we would've seen it in there